arangodb-http/src/main/scala/io/funkode/arangodb/http/ArangoClientHttp.scala

```diff
@@ -123,12 +123,24 @@ class ArangoClientHttp[Encoder[_], Decoder[_]](
   def withConfiguration(newConfig: ArangoConfiguration): ArangoClient[Encoder, Decoder] =
     new ArangoClientHttp[Encoder, Decoder](newConfig, httpClient, token)
 
-  private def parseResponseBody[O: Decoder](response: Response)(using Decoder[ArangoError]): AIO[O] =
-    for body <-
+  private def parseResponseBody[O: Decoder](response: Response): AIO[O] =
         if response.status.isError
-        then httpDecoder.decode[ArangoError](response.body).flatMap[Any, ArangoError, O](r => ZIO.fail(r))
+        then
+         httpDecoder.decode[ArangoError](response.body) 
+         .foldCauseZIO({
+          case e =>
+            val error = response.body.asString
+              .foldZIO(e => ZIO.succeed(ArangoError(500, true, "Can't get body of response from Arango: " + e.getMessage, -1)), body =>
```

Comment:
this line is little bit long

Reply:
Do you want to force error 500 because we can't parse the error right?
I'm not sure if I would still prefer response.status.code here but it has sense your approach also

In theory we should never reach this code right?
The idea is anytime we can't parse the ArangoError we will fix the code

Reply:
why we use succeed and not ZIO.fail?